lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -8633,14 +8667,12 @@ void SelectionDAGISel::LowerArguments(const Function &F) {
     for (unsigned Value = 0, NumValues = ValueVTs.size();
          Value != NumValues; ++Value) {
       EVT VT = ValueVTs[Value];
-      Type *ArgTy = VT.getTypeForEVT(*DAG.getContext());
       ISD::ArgFlagsTy Flags;
 
-      // Certain targets (such as MIPS), may have a different ABI alignment
-      // for a type depending on the context. Give the target a chance to
-      // specify the alignment it wants.
-      unsigned OriginalAlignment =
-          TLI->getABIAlignmentForCallingConv(ArgTy, DL);
+      // Certain targets (such as MIPS or AMDGPU), may have a different ABI
+      // alignment for a type depending on the context. Give the target a chance
+      // to specify the alignment it wants.
+      unsigned OriginalAlignment = Alignments[Value];
```

Comment:
Could we guard this with IsAMDGPUKernel so that the new logic doesn't affect other calling conventions?

Reply:
`Alignments[]` should already contain proper alignment information for given target / ABI. I think it's the comment may have to be moved around.